src/ui/views/session/SessionView.tsx

```diff
@@ -37,7 +38,8 @@ export function SessionView() {
   const { worldId, setInitialWorld, leftWorld, enteredWorld } = useStore((state) => state.world);
   const world = useRoom(session, worldId);
 
-  const nextWorldId = worldMatch ? worldMatch.params["worldId"] || location.hash : undefined;
+  const nextWorld = getRoomWithAlias(session.rooms, location.hash);
+  const nextWorldId = worldMatch ? worldMatch.params["worldId"] : nextWorld?.id;
```

Comment:
Is there a case where `nextWorld` is undefined and so `nextWorldId` is neither a room id or room alias? For example if you navigate to `https://thirdroom.io/world/#somenewworld:matrix.org` and you haven't joined that room yet. How do we handle this?

Reply:
if alias does not have room we will have nextWorld as `undefined`